src/_pytest/_code/source.py

```diff
@@ -282,7 +283,7 @@ def compile_(  # noqa: F811
     return s.compile(filename, mode, flags, _genframe=_genframe)
 
 
-def getfslineno(obj) -> Tuple[Optional[Union["Literal['']", py.path.local]], int]:
+def getfslineno(obj: Any) -> Tuple[Optional[Union[str, py.path.local]], int]:
```

Comment:
@bluetech 
Not sure about `obj: Any` here.  I've tried `Callable[[Any], None]` inbetween, but that would e.g. not cover classes etc.
I think it makes sense to allow Any here explicitly, given that TypeErrors are handled etc anyway.

Reply:
Yes, looks like this really is supposed to handle everything. I'd make it `object` though, instead of `Any`. `object` means the code can assume *nothing* on the value, while `Any` means the code can assume *anything*.

Reply:
Then it complains with the `inspect.get{source,}file` calls though, which expects `"Union[Module, Type[Any], MethodType, FunctionType, TracebackType, FrameType, CodeType, Callable[..., Any]]"`.
Should we use this then maybe also, or `# type: ignore` it?

Reply:
Catching `TypeError` is basically a runtime `type: ignore`. So seems justifiable in this case. 

(Of course, eventually it will be good to get rid of this `except TypeError`, which indicates some type confusion)